thinc/backends/ops.py

```diff
@@ -636,13 +636,14 @@ def as_contig(self, data: ArrayT, dtype: Optional[DTypes] = None) -> ArrayT:
         return self.xp.ascontiguousarray(data, **kwargs)
 
     def sigmoid(self, X: FloatsType, *, inplace: bool = False) -> FloatsType:
-        if inplace:
-            self.xp.exp(-X, out=X)
-            X += 1.0  # type: ignore
-            X **= -1.0  # type: ignore
-            return cast(FloatsType, X)
-        else:
-            return cast(FloatsType, 1.0 / (1.0 + self.xp.exp(-X)))
+        with numpy.errstate(over="ignore"):
```

Comment:
We also do this with some other activations (although too late to avoid warnings 😅).

Reply:
Fair point, but then we'll need to pick an arbitrary range that we consider as "valid" input, no? I feel that in the specific case of `sigmoid`, `+/-inf` naturally describes that range and any clamping that we perform beforehand will functionally result in the same final output albeit with more overhead.

Reply:
For the clamping value. Do we really care about the difference between say 1e-8 or 1e-9?

Besides that clamping has the additional benefit of adding a form of regularization, there is no need to grow weight extremely to chase the asymptote.

Clamping is done in quite some implementations, eg. commonly when training word embeddings and the TF reference implementation.

Reply:
+-inf is also quite arbitrary, since it is determined by the floating point precision.

I am fine with the PR solution, but since we do clamping for other activations and has its own benefits , I think it’s word considering.

Reply:
Ah, good point about regularization. I see that we're clamping to `+/-20.0` in `softmax_sequence` before exponentiation - I'll use the same here.